filters/filter.go

```diff
@@ -65,6 +68,21 @@ var (
 	multiRangeSplit = []byte(",")
 )
 
+// FilterValue contains the filter pattern and a boolean flag indicating
+// whether the filter should be negated.
+type FilterValue struct {
+	Pattern string
+	Negate  bool
```

Comment:
What do you think of making this a more general `Flags` byte and checking if a bit set to determine is a flag is enabled? Then we would be able to add up to seven more flags in the future if we need to without having to change the struct.

Reply:
Any particular extension you have in mind? Adding more flags would require making similar changes to the protobuf as well so I'd like to know about the usecases first.

Reply:
I was just thinking that we may want to support regex in the future, even if we don't run it on our own servers. Then, rather than adding a new bool, we could just check another bit in the bitflag.

Reply:
I see, in that case I'd actually just add a `UseRegex` flag to make it more explicit rather than manipulating bits. The issue w/ a bit flag is that now you need to do `IsNegate` checks in the code every where and handle error cases, which makes code messier.